File: show_metadata.py
from pathlib import Path
import datetime

def show_metadata(file_path, config):
    if not check_file_permission(file_path, "read", config):
        raise PermissionError(f"Read access denied for {file_path}")
    file_path = Path(file_path).resolve()
    if not file_path.is_file():
        raise ValueError(f"File not found: {file_path}")
    stat = file_path.stat()
    size = stat.st_size
    mtime = datetime.datetime.fromtimestamp(stat.st_mtime).isoformat()
    return f"File: {file_path}\nSize: {size} bytes\nModified: {mtime}"

File: ts-tools/show_metadata.ts
/**
 * @license
 * Copyright 2025 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

import * as fs from 'fs';
import * as path from 'path';
import { checkFilePermission } from './check_file_permission';

export function showMetadata(filePath: string, config: any): Promise<string> {
  return new Promise((resolve, reject) => {
    checkFilePermission(filePath, 'read', config)
      .then(() => {
        const resolvedPath = path.resolve(filePath);
        fs.stat(resolvedPath, (err, stats) => {
          if (err) {
            reject(err);
          } else {
            const size = stats.size;
            const mtime = stats.mtime.toISOString();
            resolve(
              `File: ${resolvedPath}\nSize: ${size} bytes\nModified: ${mtime}`,
            );
          }
        });
      })
      .catch(reject);
  });
}
